### ISSUE TITLE
Confirmation email sent when user signs in or out with Mongoid 5

### ISSUE THREAD

**Issue by dfabreguette-ap** [plataformatec/devise#4160]
Looks like the "after_commit" callback is now defined in Mongoid 5. After an upgrade, I had confirmation emails sent out even though user account was confirmed (confirmed? return true).
Problem seems to come from the following callback :

```
after_commit :send_reconfirmation_instructions, on: :update, if: :reconfirmation_required?
```

Defined by mistake because the "after_commit" check.

I pushed a fork to fix that, doing the check on the ActiveRecord::Base definition.

https://github.com/plataformatec/devise/pull/4159/files

**Comment by nazarok** [comment]
mongoid-5.1.6 as well as mongoid-6.0.3 not have"after_commit" callback!
But gem 'mongoid-paperclip' defined it here https://github.com/mrrooijen/mongoid-paperclip/blob/master/lib/mongoid_paperclip.rb#L68.
So the PR solve this problem.
+1

**Comment by rodrigobdz** [comment]
Any updates on this one? @tegon

**Comment by tegon** [comment]
@rodrigobdz The PR still needs a rebase with master and a test case.

**Comment by tegon** [comment]
Fixed on #4302
